File: apps/web/src/types/resume.ts
export interface Link {
  name: string
  url: string
}

export interface ContactInfo {
  full_name: string
  email?: string | null
  phone?: string | null
  address?: string | null
  links: Link[]
}

export interface WorkExperience {
  company: string
  position: string
  start_date?: string | null
  end_date?: string | null
  is_current?: boolean
  description?: string | null
  responsibilities: string[]
  skills: string[]
}

export interface Education {
  institution: string
  degree: string
  field_of_study?: string | null
  graduation_date?: string | null
  gpa?: number | null
  honors: string[]
  relevant_courses: string[]
  skills: string[]
}

export interface Certification {
  name: string
  issuer: string
  date_obtained?: string | null
  expiration_date?: string | null
  credential_id?: string | null
}

export interface Resume {
  contact_info: ContactInfo
  summary?: string | null
  work_experience: WorkExperience[]
  education: Education[]
  certifications: Certification[]
  skills: string[]
}

File: apps/api/app/schema/resume.py
from typing import List, Optional

from pydantic import BaseModel


class Link(BaseModel):
    name: str
    url: str


class ContactInfo(BaseModel):
    full_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    address: Optional[str] = None
    links: List[Link] = []


class WorkExperience(BaseModel):
    company: str
    position: str
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    is_current: Optional[bool] = False
    description: Optional[str] = None
    responsibilities: List[str] = []
    skills: List[str] = []


class Education(BaseModel):
    institution: str
    degree: str
    field_of_study: Optional[str] = None
    graduation_date: Optional[str] = None
    gpa: Optional[float] = None
    honors: List[str] = []
    relevant_courses: List[str] = []
    skills: List[str] = []


class Certification(BaseModel):
    name: str
    issuer: str
    date_obtained: Optional[str] = None
    expiration_date: Optional[str] = None
    credential_id: Optional[str] = None


class Resume(BaseModel):
    contact_info: ContactInfo
    summary: Optional[str] = None
    work_experience: List[WorkExperience] = []
    education: List[Education] = []
    certifications: List[Certification] = []
    skills: List[str] = []
